helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -240,8 +248,7 @@ public long getNextRebalanceTime() {
     @Override
     public void run() {
       try {
-        if (_clusterEventType.equals(ClusterEventType.PeriodicalRebalance) || _clusterEventType
-            .equals(ClusterEventType.OnDemandRebalance)) {
+        if (_shouldRefreshCache) {
```

Comment:
Is this a behavior change (since it could be risky in different parts of the controller pipeline - as in it might refresh cache when it doesn't need to and increase latency)?

Reply:
I tried to keep everything work as the old day. Logic only is changed if the code is using the new methods.
Please let me know if you find any outliers.